lens/config.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package lens
+
+import "github.com/lens-vm/lens/host-go/engine/module"
+
+// Option is a funtion that sets a config value on the lens registry.
+type Option func(*lensRegistry)
```

Comment:
todo: Please do not define public-public stuff (i.e. not just internal-public stuff) in the lens package.  Users (or our integration tests) should not ever be referencing this package, and by adding stuff that requires the direct reference you are making the existing internal-public stuff public-public.

Please move these somewhere else, or given that they are setting private stuff, maybe have public-public funcs in `client` or `db` that call these internal-public functions, and make sure you remove the reference to this package from the integration tests.

Reply:
We should move the lens package into an `internal` sub-package if it's meant to be internal only.

Reply:
Yes, this has been a long running battle, not everyone wants an internal package, although as far as I know everyone agrees that only a couple of packages are public.  I created an issue for a proper internal package and we can discuss it in the standup - this conversation seems to happen every other month or so.

https://github.com/sourcenetwork/defradb/issues/2507

Reply:
To clarify, as far as I know, the fact that only a couple of packages are public is not really disputed by anyone - we all agree that public stuff can only live in `client` etc.  What is disputed is whether that public/internal split should be enforced/documented by an `internal` directory or not.